download-and-preprocessing.kts

```diff
@@ -97,4 +105,28 @@ fun rename(dirString: String, type: String) {
 fun snakeToCamel(snake: String): String {
     return snake.split("-")
         .joinToString("") { it.capitalize() }
+}
+
+fun deleteDirectory(directory: File): Boolean {
+    if (directory.exists()) {
+        directory.listFiles()?.forEach { file ->
+            if (file.isDirectory) {
+                deleteDirectory(file)
+            } else {
+                file.delete()
+            }
+        }
+    }
+    return directory.delete()
+}
+
+fun setSvgSizeTo24(file: File) {
+    val svgContent = file.readText()
+    // viewBox="0 0 256 256"
+    val updatedSvgContent = svgContent.replace(
+        """(viewBox="\d+ \d+ \d+ \d+")""".toRegex(),
+        "viewBox=\"0 0 24 24\""
```

Comment:
The `viewBox` size shouldn't be changed, that has to always stay `256 256`. Otherwise icons won't render properly (the `viewBox` is essentially the internal canvas used by the path commands).

What should be changed is the `defaultWidth` and `defaultHeight` parameters in the resulting Kotlin code. I suspect that has to be done _after_ the conversion to Kotlin.

Reply:
Yes, you are right.

Reply:
If we need to modify the default size, many values in a file need to be converted.
Here is a sample. 
``` kotlin
package com.adamglin.phosphoricons.regular

import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.PathFillType.Companion.NonZero
import androidx.compose.ui.graphics.SolidColor
import androidx.compose.ui.graphics.StrokeCap.Companion.Butt
import androidx.compose.ui.graphics.StrokeJoin.Companion.Miter
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.graphics.vector.ImageVector.Builder
import androidx.compose.ui.graphics.vector.path
import androidx.compose.ui.unit.dp
import com.adamglin.phosphoricons.RegularGroup

public val RegularGroup.AddressBookTabs: ImageVector
    get() {
        if (_addressBookTabs != null) {
            return _addressBookTabs!!
        }
        _addressBookTabs = Builder(name = "AddressBookTabs", defaultWidth = 256.0.dp, defaultHeight
                = 256.0.dp, viewportWidth = 256.0f, viewportHeight = 256.0f).apply {
            path(fill = SolidColor(Color(0xFF000000)), stroke = null, strokeLineWidth = 0.0f,
                    strokeLineCap = Butt, strokeLineJoin = Miter, strokeLineMiter = 4.0f,
                    pathFillType = NonZero) {
                moveTo(208.0f, 32.0f)
                lineTo(48.0f, 32.0f)
                arcTo(16.0f, 16.0f, 0.0f, false, false, 32.0f, 48.0f)
                lineTo(32.0f, 208.0f)
                arcToRelative(16.0f, 16.0f, 0.0f, false, false, 16.0f, 16.0f)
                lineTo(208.0f, 224.0f)
                arcToRelative(16.0f, 16.0f, 0.0f, false, false, 16.0f, -16.0f)
                lineTo(224.0f, 48.0f)
                arcTo(16.0f, 16.0f, 0.0f, false, false, 208.0f, 32.0f)
                close()
                moveTo(192.0f, 104.0f)
                horizontalLineToRelative(16.0f)
                verticalLineToRelative(48.0f)
                lineTo(192.0f, 152.0f)
                close()
                moveTo(208.0f, 88.0f)
                lineTo(192.0f, 88.0f)
                lineTo(192.0f, 48.0f)
                horizontalLineToRelative(16.0f)
                close()
                moveTo(48.0f, 48.0f)
                lineTo(176.0f, 48.0f)
                lineTo(176.0f, 208.0f)
                lineTo(48.0f, 208.0f)
                close()
                moveTo(208.0f, 208.0f)
                lineTo(192.0f, 208.0f)
                lineTo(192.0f, 168.0f)
                horizontalLineToRelative(16.0f)
                verticalLineToRelative(40.0f)
                close()
                moveTo(151.75f, 166.0f)
                arcToRelative(39.76f, 39.76f, 0.0f, false, false, -17.19f, -23.34f)
                arcToRelative(32.0f, 32.0f, 0.0f, true, false, -45.12f, 0.0f)
                arcTo(39.84f, 39.84f, 0.0f, false, false, 72.25f, 166.0f)
                arcToRelative(8.0f, 8.0f, 0.0f, false, false, 15.5f, 4.0f)
                curveToRelative(2.64f, -10.25f, 13.06f, -18.0f, 24.25f, -18.0f)
                reflectiveCurveToRelative(21.62f, 7.73f, 24.25f, 18.0f)
                arcToRelative(8.0f, 8.0f, 0.0f, true, false, 15.5f, -4.0f)
                close()
                moveTo(96.0f, 120.0f)
                arcToRelative(16.0f, 16.0f, 0.0f, true, true, 16.0f, 16.0f)
                arcTo(16.0f, 16.0f, 0.0f, false, true, 96.0f, 120.0f)
                close()
            }
        }
        .build()
        return _addressBookTabs!!
    }

private var _addressBookTabs: ImageVector? = null

```